File: evolution/token_counter.py
"""
Token estimation for the evolution loop.

Uses a simple heuristic (len(text) // 4) — accurate enough for trend
tracking. Not a replacement for tiktoken; never use for billing.
"""
from typing import Sequence


def estimate_tokens(text: str) -> int:
    """Estimate token count for a single string."""
    return len(text) // 4


def sum_tokens(*parts: str) -> int:
    """Sum estimated tokens across multiple strings."""
    return sum(estimate_tokens(p) for p in parts)

File: src/token-counter.ts
/**
 * Token estimation for the evolution loop.
 *
 * Uses a simple heuristic (Math.floor(text.length / 4)) — symmetric with
 * evolution/token_counter.py. Not tiktoken-accurate; for trend tracking only.
 */

export function estimateTokens(text: string): number {
  return Math.floor(text.length / 4);
}

export function sumTokens(...parts: string[]): number {
  return parts.reduce((acc, p) => acc + estimateTokens(p), 0);
}
